bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/NoboHubBridgeHandler.java

```diff
@@ -0,0 +1,425 @@
+
+import java.time.Duration;
+import java.util.Collection;
+
+import javax.validation.constraints.NotNull;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.nobohub.internal.connection.HubCommunicationThread;
+import org.openhab.binding.nobohub.internal.connection.HubConnection;
+import org.openhab.binding.nobohub.internal.discovery.NoboThingDiscoveryService;
+import org.openhab.binding.nobohub.internal.model.Component;
+import org.openhab.binding.nobohub.internal.model.ComponentRegister;
+import org.openhab.binding.nobohub.internal.model.Hub;
+import org.openhab.binding.nobohub.internal.model.NoboCommunicationException;
+import org.openhab.binding.nobohub.internal.model.NoboDataException;
+import org.openhab.binding.nobohub.internal.model.OverrideMode;
+import org.openhab.binding.nobohub.internal.model.OverridePlan;
+import org.openhab.binding.nobohub.internal.model.OverrideRegister;
+import org.openhab.binding.nobohub.internal.model.SerialNumber;
+import org.openhab.binding.nobohub.internal.model.Temperature;
+import org.openhab.binding.nobohub.internal.model.WeekProfile;
+import org.openhab.binding.nobohub.internal.model.WeekProfileRegister;
+import org.openhab.binding.nobohub.internal.model.Zone;
+import org.openhab.binding.nobohub.internal.model.ZoneRegister;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NoboHubBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ * @author Espen Fossen - Initial contribution
+ */
+@NonNullByDefault
+public class NoboHubBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(NoboHubBridgeHandler.class);
+
+    private @Nullable NoboHubBridgeConfiguration config;
+    private @Nullable HubCommunicationThread hubThread;
+    private @Nullable NoboThingDiscoveryService discoveryService;
+    private @Nullable Hub hub;
+
+    private final @NotNull OverrideRegister overrideRegister = new OverrideRegister();
+    private final @NotNull WeekProfileRegister weekProfileRegister = new WeekProfileRegister();
+    private final @NotNull ZoneRegister zoneRegister = new ZoneRegister();
+    private final @NotNull ComponentRegister componentRegister = new ComponentRegister();
+    private final NoboHubTranslationProvider messages;
```

Comment:
Unused, this compiler warning:
`NoboHubBridgeHandler.java:[79,46] The value of the field org.openhab.binding.nobohub.internal.NoboHubBridgeHandler.messages is not used`